cmd/server/authentication/authentication.go

```
@@ -57,57 +101,71 @@ func AuthenticationServerMiddleware(cfg *config.AuthenticationServerConfig, next
+				}
+				logger.L().Debug("authenticating incoming connection",
 					helpers.Int("accessKey (length)", len(accessKey)),
 					helpers.String("account", account),
 					helpers.String("cluster", cluster),
 					helpers.String("connId", connectionId),
-					helpers.Int("statusCode", response.StatusCode))
-				w.WriteHeader(http.StatusUnauthorized)
-				return
+					helpers.String("url", u.String()))
+
+				response, err := client.Do(authenticationRequest)
```

golangci-lint (2.11.4)

[error] response body must be closed

(bodyclose)